### ISSUE TITLE
Autocomplete Paired Glyphs

### ISSUE THREAD

**Issue by goossens** [santaclose/ImGuiColorTextEdit#14]
In my old fork of ImGuiColorTextEdit, I had a feature to autocomplete paired glyphs. So when typing [, {, (, “ or ‘, the matching glyph was also added. Undo removed both characters. The feature is configurable and the default is off.

I have ported this feature to this fork and you can see my changed at https://github.com/goossens/ImGuiColorTextEdit/commit/66d8c5987cbe1e5dd7b3031ac65ce15a3347a207. Let me know if you want a put request.

Let me know if you want a pull request.

**Comment by santaclose** [comment]
yeah I'm skeptical about this because it has happened to me that editors trying to be smart end up being annoying. Like, typing closing bracket first can make it add an extra closing bracket. It can definitely save time when coding though. I'll see if I can test your implementation. Related to this, maybe we should also surround selection if pressing open bracket key with text selected

**Comment by aeris170** [comment]
> yeah I'm skeptical about this because it has happened to me that editors trying to be smart end up being annoying. Like, typing closing bracket first can make it add an extra closing bracket. It can definitely save time when coding though. I'll see if I can test your implementation. Related to this, maybe we should also surround selection if pressing open bracket key with text selected

Some editors handle this by nullifying the typing of closing bracket if it's already just been typed automatically.

**Comment by goossens** [comment]
@aeris170:

Instead of maintaining a port of this repository, I have completely rewritten the editor while maintaining 99% of the public API. You can find it in my repository at https://github.com/goossens/ImGuiColorTextEdit. It has more features including nullifying closing bracket if it's already just been typed. It also has decent documentation.

**Comment by santaclose** [comment]
@aeris170 
> Some editors handle this by nullifying the typing of closing bracket if it's already just been typed automatically.

That can be annoying if you want to insert a new closing bracket

**Comment by aeris170** [comment]
> [@aeris170](https://github.com/aeris170)
> 
> > Some editors handle this by nullifying the typing of closing bracket if it's already just been typed automatically.
> 
> That can be annoying if you want to insert a new closing bracket

If you want to insert a new one, type a new one. I don't understand how that is annoying. Think of it like this:

```cpp
|
```
You type opening bracket
```cpp
[|]
```
You now type closing bracket
```cpp
[]|
```
If you want to type another closing bracket
```cpp
[]]|
```

It works exactly this way with all the IDE's I've used so far.

**Comment by santaclose** [comment]
yeah exactly, typing more than one closing bracket creates a single one. Annoying in some cases. But the editor should be configurable and allow this, yes.
